File: 611-valid-triangle-number/611-valid-triangle-number.py
class Solution(object):
    def triangleNumber(self, nums):
        answer = 0

        nums.sort()

        for i in range(2, len(nums)):
            left = 0
            right = i - 1

            while left < right:
                if nums[left] + nums[right] > nums[i]:
                    answer += right - left
                    right -= 1
                else:
                    left += 1

        return answer

File: 611-valid-triangle-number/611-valid-triangle-number.swift
class Solution {
    func triangleNumber(_ nums: [Int]) -> Int {
        var answer: Int = 0
        
        if nums.count < 3 {
            return 0
        }
        
        let numsSort: [Int] = nums.sorted()
        
        for i in 2..<numsSort.count {
            var left: Int = 0
            var right: Int = i - 1
            
            while left < right {
                if numsSort[left] + numsSort[right] > numsSort[i] {
                    answer += right - left
                    right -= 1
                } else {
                    left += 1
                }
            }
        }
        
        return answer
    }
}